File: JWP_API/JadeWesserPort/Domain/Entities/OperationalWindow.cs
namespace JadeWesserPort.Domain;

public class OperationalWindow
{
    public Guid Id { get; set; }
    public string Code { get; set; } = string.Empty;
    public DayOfWeek DayOfWeek { get; set; }

    public TimeOnly StartTime { get; set; }
    public TimeOnly EndTime   { get; set; }

    public Guid StaffMemberId { get; set; }
    public StaffMember StaffMember { get; set; } = null!;
    
}

File: Frontend/src/domain/Types/entities/OperationalWindow.ts
import type { DAYS_OF_WEEK } from "../../Enums/daysOfWeek";

export type DayOfWeek = (typeof DAYS_OF_WEEK)[number];

export type OperationalWindow = {
  code: string;
  dayOfWeek: DayOfWeek;
  startTime: string;
  endTime: string;
};
